cmd/syncthing/connections.go

```diff
@@ -283,3 +286,16 @@ func setTCPOptions(conn *net.TCPConn) {
 		l.Infoln(err)
 	}
 }
+
+func shouldLimit(addr net.Addr) bool {
```

Comment:
Indeed. I may be too pedantic, but it feels like spooky action at a distance. Can we add an explicit `if` somewhere to make this clearer (since it fooled me... ;).